src/Traverser.h

```diff
@@ -49,7 +49,7 @@ class Traverser {
     Visitor v;
     for (int i = 0; i < tp->interactions.size(); i++) {
       for (Node<Data>* source : tp->interactions[i]) {
-        v.leaf(*source, *(tp->leaves[i]));
+        if (source->key != tp->leaves[i]->key) v.leaf(*source, *(tp->leaves[i]));
```

Comment:
A comment adding an explanation for this `if` might be worthwhile. I'm unsure why it's necessary now compared to before. Assuming it is to prevent a `source` from visiting itself or something like that?

Reply:
we moved to doing particle-bucket interactions instead of particle-particle interactions in the leaf. This will blow up if you're doing a leaf-leaf interaction on yourself. This is a good point though. Maybe some users will want to? Open to a discussion

Reply:
Got it, that makes sense. When would someone want these self, leaf-leaf interactions? If there is a specific use-case in mind, it may be worth making it an option; otherwise, I think a comment would suffice for now.